src/tlo/analysis/probability_premature_death.py

```diff
@@ -0,0 +1,107 @@
+"""
+Read in the output files generated by analysis_scenarios
+generate life tables to estimate probability of premature death (defined as before age 70) for each run/draw
+produce summary statistics
+"""
+
+
+import datetime
+from pathlib import Path
+from typing import Dict, Tuple
+
+import pandas as pd
+
+from tlo.analysis.life_expectancy import _aggregate_person_years_by_age, _num_deaths_by_age_group
+from tlo.analysis.utils import get_scenario_info, summarize
+
+
+def _calculate_probability_of_dying_before_70(
+    _person_years_at_risk: pd.Series,
+    _number_of_deaths_in_interval: pd.Series
+) -> Dict[str, float]:
+    """
+    For a single run, estimate the probability of dying before age 70 for males and females.
+    Returns: Dict (keys by "M" and "F" for the sex, values the estimated probability of dying before age 70).
+    """
+    probability_of_dying_before_70 = dict()
+
+    age_group_labels = _person_years_at_risk.index.get_level_values('age_group').unique()
```

Comment:
as this logic has so much in common with the life-expectancy calculation, I think it would make sense to factor it out, in order that the same calculation can serve both purposes (life-expectnacy and probability of premature death).

I would also recommend putting these functions into the same file as the life-expectnacy calculations

Reply:
So do you think this file should go entirely, and just include it all in the life expectancy file?

Reply:
yes, I think that would be fine. 

Main point is about the refactoring to avoid repeating the logic about computing death rates in age-groups and building the cumulative probability of death. 

(Location of the function in which file is secondary, but I think it being in one file would be fine).

Reply:
Done! REfactored all logic around probability of dying